includes/class-woocommerce.php

```diff
@@ -559,13 +580,13 @@ public function init() {
 		$this->load_plugin_textdomain();
 
 		// Load class instances.
-		$this->product_factory                     = new WC_Product_Factory();
-		$this->order_factory                       = new WC_Order_Factory();
-		$this->countries                           = new WC_Countries();
-		$this->integrations                        = new WC_Integrations();
-		$this->structured_data                     = new WC_Structured_Data();
-		$this->deprecated_hook_handlers['actions'] = new WC_Deprecated_Action_Hooks();
-		$this->deprecated_hook_handlers['filters'] = new WC_Deprecated_Filter_Hooks();
+		$this->product_factory                     = $this->container->get( WC_Product_Factory::class );
```

Comment:
I think we should leave these alone. A better pattern might be to replace the contents of the singletons themselves with composed calls to other, better tested, classes.

Reply:
Sure, the classes themselves should be turned into stubs to better classes; but see my comment about old and new code interacting.